### PR TITLE
Allow editing of Workspace names

### PR DESCRIPTION
**Pull request by HorusGoul**
### Details
Adds an edit button at the right side of the workspace name that opens a modal where you can change the workspace name.

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/171962

### Tests/QA Steps
1. Create or open a workspace
2. Click the shown in along with the avatar
3. A modal with an avatar editor and a name input should open. 
4. In that modal, change everything and press "Save"
5. The modal should close, and everything you changed in the modal should get updated everywhere.

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web

https://user-images.githubusercontent.com/6759612/127823948-a77f5fc0-4565-4892-a774-1169c49b2aa1.mov

#### Mobile Web

https://user-images.githubusercontent.com/6759612/127824562-43a49165-7921-430a-ae07-5e24f59e9bda.mov

#### Desktop

https://user-images.githubusercontent.com/6759612/127823948-a77f5fc0-4565-4892-a774-1169c49b2aa1.mov

#### iOS

https://user-images.githubusercontent.com/6759612/127823077-9a68475e-4371-417c-8333-ba92ff8d3a27.mov

#### Android

https://user-images.githubusercontent.com/6759612/127823730-836191ae-aceb-4b45-83d4-975cf1da20b2.mov

### CODE REVIEW

**Comment by kevinksullivan** [comment]
@HorusGoul @MitchExpensify I don't think we want to add another pencil icon. Let's keep the one pencil icon, and then when a user selects that we open this right pane, which allows a user to edit their workspace avatar OR their workspace name.


image

**Comment by mountiny** [comment]
Was just testing it and I agree with @kevinksullivan, that will be cleared, two same icons right next to each other seem weird.

**Comment by kevinksullivan** [comment]
@shawnborton I am also curious for your thoughts on this one. I think we've discussed the idea of a business overview page for the purpose of editing name, avatar, business  address, etc... But in the meantime, I think it is ok to stick with a single pencil icon that opens the right pane for editing avatar and/or name.

**Comment by shawnborton** [comment]
I agree with that sentiment Kevin. 

Separately, I wonder if you wouldn't mind making some quick tweaks to the pencil icon/circle that's over the workspace icon? Ideally it would look more like this:
image


- The gray circle is 28x28
- The pencil icon is 20x20
- The circle is -4 from the bottom edge and -4 from the right edge

**Comment by HorusGoul** [comment]
Awesome, I'll move the avatar editor inside the modal and leave only 1 pencil icon.

> Separately, I wonder if you wouldn't mind making some quick tweaks to the pencil icon/circle that's over the workspace icon?

Sure!

**Comment by MitchExpensify** [comment]
Completely agree after seeing the mock!

**Review by mountiny — DISMISSED** [review]
Haven't spotted any blockers in the code! Tested and works well 🎉 Great job!

**Comment by HorusGoul** [comment]
> Haven't spotted any blockers in the code! Tested and works well 🎉 Great job!

I just found this little issue regarding the pencil button on iOS. I think it's related to the border being drawn inside the circle instead of outside:

image

Maybe that's why we had the other width/height values before. I'm gonna send another commit with a fix.

**Comment by HorusGoul** [comment]
It should be ready for review again, and all videos have been updated in case you want to take a look!

**Review by mountiny — DISMISSED** [review]
Thanks for catching that small problem. LGTM

**Comment by shawnborton** [comment]
Looks good, thanks!

File: src/pages/workspace/WorkspaceEditorPage.js [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+                            onImageRemoved={this.onImageRemoved}
+                        />
+
+                        <TextInputWithLabel
+                            label={this.props.translate('workspace.editor.nameInputLabel')}
+                            value={this.state.name}
+                            onChangeText={name => this.setState({name})}
+                            onSubmitEditting={this.submit}
+                        />
+                        <Text style={[styles.mt2]}>{this.props.translate('workspace.editor.nameInputHelpText')}</Text>
```
Comment (shawnborton) [inline src/pages/workspace/WorkspaceEditorPage.js:120]: Can we also use `fontSizeLabel` and `colorMuted` here? This way it appears like hint text under the input.

**Comment by shawnborton** [comment]
Actually after looking at this a second time, I wonder if it feels weird that you see the avatar image with a pencil icon twice. Maybe for consistency's sake, we should remove the pencil icon from the first screen and just make it so that tapping the avatar or the workspace name will launch the rightDocked modal. From here, I would expect the avatar to have the pencil icon on it. This way we stay more consistent with say editing your profile, where you have to first click on your avatar image (with no pencil icon).

**Comment by kevinksullivan** [comment]
Yeah I agree with Shawn. Showing the pencil icon twice feels like overkill here, and I like keeping user avatar and workspace avatar designs consistent.

File: src/pages/workspace/WorkspaceEditorPage.js (hunk 1 above)
Comment (HorusGoul) [reply src/pages/workspace/WorkspaceEditorPage.js]: I see there's a formHint style that has those! I just sent the commit with the changes 😄

**Comment by HorusGoul** [comment]
>  we should remove the pencil icon from the first screen and just make it so that tapping the avatar or the workspace name will launch the rightDocked modal.

Tapping the avatar or the workspace name to access the editor could look weird, as there would be no visual hints for the user. If we want to make this consistent with the user profile editor, we need to add a new menu option like this one:

image

What do you think?

**Comment by mountiny** [comment]
That is a good point, at the same time, I would say it is not very uncommon from other applications and UIs that you click on the avatar/name to change it (Facebook for example).

What I would suggest is instead of adding another `Profile` or its equivalent for Workspace, make the User Avatar and name `Pressable` too in the Settings panel and it will lead to the Profile page. 

This will ensure a consistency between these two without over-engineering the Workspace page. What do you think? 👀

**Comment by HorusGoul** [comment]
I just submitted the changes to make it that way for both the User profile and the Workspace profile. I've also updated the videos 😄

**Review by mountiny — APPROVED** [review]
Great, tested! Love this flow.

**Pull request merged**